src/Theme.php

```diff
@@ -0,0 +1,205 @@
+     * @return Theme[]
+     */
+    public static function getCustomThemes(): array
+    {
+        static $custom_themes = null;
+
+        if ($custom_themes === null) {
+            $custom_themes = [];
+
+            $file_matches = [];
+            // Cannot use GLOB_BRACE on some platforms (like the docker environment used for tests)
+            $patterns = [
+                '*.css',
+                '*.scss'
+            ];
+            foreach ($patterns as $pattern) {
+                foreach (glob(self::CORE_THEME_ROOT . $pattern) as $file) {
+                    $file_name = pathinfo($file, PATHINFO_FILENAME);
+                    if (str_starts_with($file_name, '_')) {
+                        continue;
+                    }
+                    $file_matches[$file_name] = $file;
+                }
+            }
+
+            $core_themes = self::getCoreThemes();
+            $core_keys = [];
+            foreach ($core_themes as $core_theme) {
+                $core_keys[] = $core_theme->getKey();
+            }
+
+            foreach ($file_matches as $file_name => $file) {
+                if (!in_array($file_name, $core_keys, true)) {
+                    // Guess dark mode based on the filename ("dark" with - or _ before or after)
+                    $pattern = '/dark[-_]|[-_]dark/i';
+                    $is_dark = preg_match($pattern, $file_name);
+                    $theme_name = ucfirst(str_replace('_', ' ', $file_name));
+                    $custom_themes[] = new self($file_name, $theme_name, $is_dark, true);
+                }
+            }
+        }
+        return $custom_themes;
+    }
+
+    /**
+     * Get all available themes
+     * @return Theme[]
+     */
+    public static function getAllThemes(): array
+    {
+        $core_themes = self::getCoreThemes();
+        $custom_themes = self::getCustomThemes();
+        return array_merge($core_themes, $custom_themes);
+    }
+
+    /**
+     * Get the Theme object for the current $_SESSION['glpipalette'] value.
+     *
+     * If the "glpipalette" value is not set, the defautl theme is used.
+     * @return Theme|null
+     */
+    public static function getCurrentTheme(): ?Theme
+    {
+        $current = $_SESSION['glpipalette'] ?? self::DEFAULT_THEME;
+        return self::getTheme($current);
+    }
+
+    /**
+     * Get the Theme object for the given key.
+     * @param string $key
+     * @return Theme|null
+     */
+    public static function getTheme(string $key): ?Theme
+    {
+        $themes = self::getAllThemes();
+        foreach ($themes as $theme) {
+            if ($theme->getKey() === $key) {
+                return $theme;
+            }
+        }
+        return null;
+    }
+
+    public function __construct(string $key, string $name, bool $is_dark, bool $is_custom)
+    {
+        $this->key = $key;
+        $this->name = $name;
+        $this->is_dark = $is_dark;
+        $this->is_custom = $is_custom;
+    }
```

Comment:
Constructor should probably moved to be the first method declared. Not sure it is a convention, but it is always declared at this place in GLPI classes, and also in most of projects I know.